src/haskell/install.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+set -e
+set -o xtrace
+
+GHCUP_VERSION="0.1.19.2"
+
+architecture="$(arch)"
+GHCUP_BIN="${architecture}-linux-ghcup-${GHCUP_VERSION}"
+
+BOOTSTRAP_HASKELL_GHC_VERSION="${VERSION:-"recommended "}"
+
+# Maybe install curl, gcc, make
+for x in curl gcc make; do
+    which $x > /dev/null || (apt update && apt install $x -y -qq)
+done
+
+GHCUP_DIR=${_REMOTE_USER_HOME}/.ghcup/bin
+
+mkdir -p $GHCUP_DIR
+echo https://downloads.haskell.org/~ghcup/${GHCUP_VERSION}/${architecture}-linux-ghcup-${GHCUP_VERSION} --output ${GHCUP_BIN}
+curl https://downloads.haskell.org/~ghcup/${GHCUP_VERSION}/${architecture}-linux-ghcup-${GHCUP_VERSION} --output ${GHCUP_BIN}
+# echo "25b7fc417c1a811dd7ff439b67ea647a59cf5b8d71b274f97e917d50b2150d5b ${GHCUP_BIN}" | sha256sum --check --status
+
+mv ${GHCUP_BIN} $GHCUP_DIR/ghcup
+chmod a+x $GHCUP_DIR/ghcup
+
+export GHCUP_INSTALL_BASE_PREFIX=${_REMOTE_USER_HOME}
+
+${GHCUP_DIR}/ghcup install ghc $BOOTSTRAP_HASKELL_GHC_VERSION
```

Comment:
I suggest to not do all this manually, but use the bootstrap script, which allows fine grained control via env vars.

E.g.

```sh
curl --proto '=https' --tlsv1.2 -sSf https://get-ghcup.haskell.org | BOOTSTRAP_HASKELL_NONINTERACTIVE=1 BOOTSTRAP_HASKELL_CABAL_VERSION=$CABALVERSION BOOTSTRAP_HASKELL_INSTALL_HLS=1 BOOTSTRAP_HASKELL_ADJUST_BASHRC=1 sh
```

This will also try do adjust shell config to add `~/.ghcup/bin` to PATH and most importantly install stack installation hooks that make sure HLS actually works with stack (there are ABI issues). See https://www.haskell.org/ghcup/guide/#strategy-2-stack-hooks-new-recommended

Right now the bootstrap script assumes you want latest of stack and HLS, but I can add environment variables to control that as well.

Reply:
Oh and the bootstrap script also creates a `~/.ghcup/env` file that can be sourced to get the correct PATH.